packages/ts/file-router/src/vite-plugin.ts

```diff
@@ -116,13 +117,12 @@ export default function vitePluginFileSystemRouter({
           // replacing any newly created configuration objects (`nextExports.config`)
           // with it. This ensures that the HMR mechanism perceives the
           // configuration as unchanged.
-          modifiedCode = modifiedCode.replace(
-            hmrInjectionPattern,
-            `if (!nextExports) return;
-            if (Object.keys(nextExports).length === 2 && 'default' in nextExports && 'config' in nextExports) {
-              nextExports = { ...nextExports, config: currentExports.config };
-            }`,
-          );
+          const injectionPattern = /import\.meta\.hot\.accept[\s\S]if\s\(!nextExports\)\s+return;/gu;
+          injectionPattern.test(modifiedCode);
+
+          modifiedCode = `${modifiedCode.substring(0, injectionPattern.lastIndex)}${INJECTION}${modifiedCode.substring(
```

Comment:
Testing this in the actual project results in `injectionPattern.lastIndex` being empty, the file being rewritten to
```
import {createHotContext as __vite__createHotContext} from "/VAADIN/@vite/client";
import.meta.hot = __vite__createHotContext("/views/@index.tsx");
if (Object.keys(nextExports).length === 2 && 'default'in nextExports && 'config'in nextExports) {
    nextExports = {
        ...nextExports,
        config: currentExports.config
    };
}
```
and failing runtime with
```
Uncaught ReferenceError: nextExports is not defined
```

Reply:
But if I add a `*` after `[\s\S], then it works and is fast

Reply:
Oops, sorry, silly mistake 😓